### PR TITLE
Type the return type of `@model_validator(mode="after")` with `Self`

### PR DESCRIPTION
**Pull request by cwognum**
## Changelogs

- Type the return type of `@model_validator(mode="after")` with `Self`

---

_Checklist:_

- [ ] ~_Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._~
- [ ] ~_Add tests to cover the fixed bug(s) or the newly introduced feature(s) (if appropriate)._~
- [ ] ~_Update the API documentation if a new function is added, or an existing one is deleted._~
- [X] _Write concise and explanatory changelogs above._
- [X] _If possible, assign one of the following labels to the PR: `feature`, `fix`, `chore`, `documentation` or `test` (or ask a maintainer to do it for you)._

---

This came up in #186 , but I only saw it after merging. See https://github.com/polaris-hub/polaris/pull/186#discussion_r1755296478 .

### CODE REVIEW

**Comment by cwognum** [comment]
Sorry for not waiting for your response @jstlaurent ! Quickly opened this PR to still address the feedback! :smile:

File: polaris/dataset/_competition_dataset.py [hunk 1]
```diff
@@ -13,7 +14,7 @@ class CompetitionDataset(DatasetV1):
     """
 
     @model_validator(mode="after")
-    def _validate_model(self):
+    def _validate_model(self) -> Self:
```
Comment (Andrewq11) [inline polaris/dataset/_competition_dataset.py:17]: Is `self` implicitly returned from `@model_validator`? We don't do it explicitly here so the `Self` typing may not be applicable.

**Review by Andrewq11 — APPROVED** [review]
Thanks @cwognum 🚀 

Left a single comment. Looks good to me once that is addressed.

File: polaris/dataset/_competition_dataset.py (hunk 1 above)
Comment (Andrewq11) [reply polaris/dataset/_competition_dataset.py:17]: A better way to put it, could we add `return self` at the end of this validator?

Comment (jstlaurent) [reply polaris/dataset/_competition_dataset.py:17]: The methods needs to `return self`at the end.

**Review by jstlaurent — APPROVED** [review]
👍

File: polaris/dataset/_competition_dataset.py (hunk 1 above)
Comment (cwognum) [reply polaris/dataset/_competition_dataset.py:17]: Good catch! Thanks! 🙏

**Pull request merged**